### PR TITLE
refactor: Add feg GitHub Actions Workflow 

### PR DESCRIPTION
**Pull request by jwagantall**
-Add initial feg GHA workflow.

These workflow is migrated from CircleCI.

Signed-off-by: Jessica Wagantall <jwagantall@linuxfoundation.org>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: .github/workflows/feg-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        timeout_minutes: 10
+    - name: make feg precommit
+      run: |
+          cd ${MAGMA_ROOT}/feg/gateway
+          make -C ${MAGMA_ROOT}/feg/gateway precommit
+#    - magma_slack_notify
+
+########## FEG-BUILD ##########
+
+  feg-build:
```
Comment (uri200) [inline .github/workflows/feg-workflow.yml:103]: note this flow will change on Circle CI

https://github.com/magma/magma/pull/6288/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R715

File: .github/workflows/feg-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    needs: feg-precommit
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+#      - build/determinator:
+#          paths: "cwf/k8s"
```
Comment (bramwelt) [inline .github/workflows/feg-workflow.yml]: Github Actions support limiting workflows to changes to [specific files](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) on pushes and pull requests (though not tags).

I'm not certain if both these jobs are supposed to be triggered when changes happen to files under `cwf/k8s`, or if only _this_ job should be triggered. If it's the latter, then you may need to look into using a [label action](https://github.com/marketplace/actions/labeler) to signify a PR made changes to a file, followed by an action that's triggered on those [labels](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#label).

File: .github/workflows/feg-workflow.yml (hunk 1 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml:103]: Thanks for this. Let me update my PR

File: .github/workflows/feg-workflow.yml (hunk 2 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: Thanks Trevor!.. I just double checked and all feg jobs should be triggered for paths under *federated_build_verify

In that case, should I still look into the label action?

Comment (bramwelt) [reply .github/workflows/feg-workflow.yml]: If it's all the jobs, then you should be able to trigger on the [paths](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) and not worry about labeling.

File: .github/workflows/feg-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+########## FEG-BUILD ##########
+
+  feg-build:
+
+    needs: feg-precommit
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+#      - build/determinator:
```
Comment (tmdzk) [inline .github/workflows/feg-workflow.yml]: Can you remobe that?

File: .github/workflows/feg-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,179 @@
+name: feg-workflow
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```
Comment (tmdzk) [inline .github/workflows/feg-workflow.yml:15]: can we add that: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

File: .github/workflows/feg-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            if [ ! -z $PROJECT ]; then
+              IMAGE_TOSEARCH="${PROJECT}_${IMAGE}"
+            fi
+            IMAGE_ID=$(docker images "$IMAGE_TOSEARCH:latest" --format "{{.ID}}")
+            tag_and_push "$TAG"
+            if [ "$TAG_LATEST" = true ]; then
+              tag_and_push "latest"
+            fi
+          done
+    - name: tag-push-docker
```
Comment (tmdzk) [inline .github/workflows/feg-workflow.yml]: redundant no?

**Review by tmdzk — CHANGES_REQUESTED** [review]

**Comment by tmdzk** [comment]
Also can you commit something to re-trigger the DCO check?

File: .github/workflows/feg-workflow.yml (hunk 5 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: Yea,, could be redundant but the tag and push is done in two different registries. 
This is where I wish Anchors could help out so we don't have to re-write code :(

**Review by jwagantall — COMMENTED** [review]
@tmdzk  thanks for your comments. I am updating this workflow now since it was missing some of the parts with regards to the build determinator for example. 

Let me update it again and it should trigger the DCO

**Comment by jwagantall** [comment]
@tmdzk , on regards to your question about the redundancy with the docker push and tag, the original workflow does a push using the docker registry and a 2nd push to the Jfrog registry. This is why I am adding the 2 pushes. This is an example I was having issues with becuase we can't use anchors, we need to copy the whole step twice (one per registry)

File: .github/workflows/feg-workflow.yml [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            ./ci-scripts/tag-push-docker.sh -i 'gateway_go|gateway_python' -t '${GITHUB_SHA:0:8}' -tl false -p feg
+      - name: persist-githash-version - Set prefix
+        run: |
+            echo "file_prefix=feg" >> $GITHUB_ENV
+      - name: persist-githash-version - Create version file
+        run: |
+            cd ${MAGMA_ROOT}/circleci
+            mkdir -p versions
+            echo "${GITHUB_SHA:0:8}" > versions/<< ${file_prefix} >>_version
+      # Notify ci channel when failing
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml:152]: There is one step missing where you notify the ci-artifact slack channel that the artifact was correctly published

File: .github/workflows/feg-workflow.yml [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            # create snowflake
+            cd ${MAGMA_ROOT}/.cache/feg/ && touch snowflake
+      - name: Export Docker registry
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway/docker
+            echo "DOCKER_REGISTRY=feg_ docker-compose build --parallel" >> $GITHUB_ENV
+      - name: run docker containers and check health
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway
+            echo "DOCKER_REGISTRY=feg_ make -C ${MAGMA_ROOT}/feg/gateway docker_healthcheck" >> $GITHUB_ENV
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: Looks like it shoudl be this instead
`DOCKER_REGISTRY=feg_ make -C ${MAGMA_ROOT}/feg/gateway docker_healthcheck`

File: .github/workflows/feg-workflow.yml [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+            # create test certs
+            cd ${MAGMA_ROOT}/.cache/test_certs/
+            openssl genrsa -out rootCA.key 2048
+            openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 365000 -out rootCA.pem -subj "/C=US/CN=rootca.magma.test"
+            # create snowflake
+            cd ${MAGMA_ROOT}/.cache/feg/ && touch snowflake
+      - name: Export Docker registry
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway/docker
+            echo "DOCKER_REGISTRY=feg_ docker-compose build --parallel" >> $GITHUB_ENV
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: Looks like it should be this instead 
`DOCKER_REGISTRY=feg_ docker-compose build --parallel`

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
See my comments about the missing slack notifications and feg build steps

File: .github/workflows/feg-workflow.yml [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+      - name: run docker containers and check health
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway
+            echo "DOCKER_REGISTRY=feg_ make -C ${MAGMA_ROOT}/feg/gateway docker_healthcheck" >> $GITHUB_ENV
+      - name: Tag and push to Docker Registry
+        run: |
+            ./ci-scripts/tag-push-docker.sh -i 'gateway_go|gateway_python' -t '${GITHUB_SHA:0:8}' -tl false -p feg
+      - name: persist-githash-version - Set prefix
+        run: |
+            echo "file_prefix=feg" >> $GITHUB_ENV
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: don't need one step to export file_prefix, just use it direclty below

File: .github/workflows/feg-workflow.yml (hunk 8 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: Updated to the latest master https://github.com/magma/magma/pull/7836/files

File: .github/workflows/feg-workflow.yml (hunk 9 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml (hunk 7 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: Updated to the latest master https://github.com/magma/magma/pull/7836/files

File: .github/workflows/feg-workflow.yml (hunk 6 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml:152]: added

File: .github/workflows/feg-workflow.yml [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+            # create test certs
+            cd ${MAGMA_ROOT}/.cache/test_certs/
+            openssl genrsa -out rootCA.key 2048
+            openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 365000 -out rootCA.pem -subj "/C=US/CN=rootca.magma.test"
+            # create snowflake
+            cd ${MAGMA_ROOT}/.cache/feg/ && touch snowflake
+      - name: Export Docker registry
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway/docker
+            echo "DOCKER_REGISTRY=feg_ python3 build.py" >> $GITHUB_ENV
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: This should be this command.

```
      - name: Export Docker registry
        env:
          DOCKER_REGISTRY: feg_
        run: |
            cd ${MAGMA_ROOT}/feg/gateway/docker
            python3 build.py
```

File: .github/workflows/feg-workflow.yml [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+      - name: run docker containers and check health
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway/docker
+            echo "DOCKER_REGISTRY=feg_ python3 build.py -e" >> $GITHUB_ENV
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: This should be this command.

```
      - name: Export Docker registry
        env:
          DOCKER_REGISTRY: feg_
        run: |
            cd ${MAGMA_ROOT}/feg/gateway/docker
            python3 build.py -e
```

File: .github/workflows/feg-workflow.yml [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+      - name: Tag and push to Docker Registry
+        run: |
+            ./ci-scripts/tag-push-docker.sh -i 'gateway_go|gateway_python' -t '${GITHUB_SHA:0:8}' -tl false -p feg
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: We have to call this script twice one with the default docker and one with the jfrog docker credentials.
Use step specific environment variable to propagate them

File: .github/workflows/feg-workflow.yml [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+      - name: persist-githash-version - Create version file
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: There is no need to persist the githash version, you need to include the version in the successful slack message.
See here for content of message: https://github.com/magma/magma/blob/ee7c301e4216a61ca32e0ed701194a265c317913/.circleci/config.yml#L503

File: .github/workflows/feg-workflow.yml [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
+          SLACK_TITLE: "Github action feg-precommit update failed"
+          SLACK_USERNAME: "Feg workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+  feg-build:
+    needs: feg-precommit
+    runs-on: ubuntu-latest
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml:105]: add this to avoid uploading builds from PRs: `if: github.ref == 'refs/heads/master'`

File: .github/workflows/feg-workflow.yml [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+          SLACK_TITLE: "Github action feg-precommit update failed"
+          SLACK_USERNAME: "Feg workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+      # Notify ci channel when push succeeds
+      - name: Notify success to slack
+        if: success() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: You must specify this WEBHOOK secret: SLACK_WEBHOOK_OSS
We are publishing successful artifact upload to ci-artifact channel

Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: Ah!.. thank you! let me fix that.

File: .github/workflows/feg-workflow.yml (hunk 10 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml (hunk 11 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

**Comment by jwagantall** [comment]
@quentinDERORY . .I might need your help on the attachment part of the Slack message.. do you have an example I can see?

Also,, i will modify the triggers for now so I can see this workflow running. I will remove the paths but I will make sure I add them later when the workflow is ready

**Comment by jwagantall** [comment]
@quentinDERORY .. finally! 
here is a better run: https://github.com/magma/magma/runs/3005544313?check_suite_focus=true

File: .github/workflows/feg-workflow.yml [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+      - name: Tag and push to Docker Registry
+        run: |
+            ./ci-scripts/tag-push-docker.sh -i 'gateway_go|gateway_python' -t '${GITHUB_SHA:0:8}' -tl false -p feg
+      - name: Tag and push to Jfrog Registry
+        env:
+          DOCKER_REGISTRY: "${{ secrets.JFROG_DOCKER_ORC8R_REGISTRY }}"
+          DOCKER_USERNAME: "${{ secrets.JFROG_USERNAME }}"
+          DOCKER_PASSWORD: "${{ secrets.JFROG_PASSWORD }}"
+        run: |
+            ./ci-scripts/tag-push-docker.sh -i 'gateway_go|gateway_python' -t '${GITHUB_SHA:0:8}' -tl false -p feg
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: Use long options name like --tag-latest true instead of just -tl
And actually switch the tag-latest to true

File: .github/workflows/feg-workflow.yml [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+            # TODO add rootCA.pem and snowflake files in the ubuntu-1604:201903-01 image
+            # create directories
+            cd ${MAGMA_ROOT} && mkdir -p .cache/test_certs/ && mkdir -p .cache/feg/
+            # create test certs
+            cd ${MAGMA_ROOT}/.cache/test_certs/
+            openssl genrsa -out rootCA.key 2048
+            openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 365000 -out rootCA.pem -subj "/C=US/CN=rootca.magma.test"
+            # create snowflake
+            cd ${MAGMA_ROOT}/.cache/feg/ && touch snowflake
+      - name: Export Docker registry
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: This step is currently called 'Build docker', maybe keep it that way or name it 'Build docker images'

File: .github/workflows/feg-workflow.yml [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
+          SLACK_TITLE: "Github action feg-precommit update failed"
+          SLACK_USERNAME: "Feg workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+      # Notify ci channel when push succeeds
+      - name: Notify success to slack
+        if: success() && github.ref == 'refs/heads/master'
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml:166]: don't need to specify `github.ref == 'refs/heads/master'` as the job will only run on master

File: .github/workflows/feg-workflow.yml [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+        env:
+          DOCKER_REGISTRY: "${{ secrets.JFROG_DOCKER_ORC8R_REGISTRY }}"
+          DOCKER_USERNAME: "${{ secrets.JFROG_USERNAME }}"
+          DOCKER_PASSWORD: "${{ secrets.JFROG_PASSWORD }}"
+        run: |
+            ./ci-scripts/tag-push-docker.sh -i 'gateway_go|gateway_python' -t '${GITHUB_SHA:0:8}' -tl false -p feg
+      # Notify ci channel when failing
+      # Plugin info: https://github.com/marketplace/actions/slack-notify
+      - name: Notify failure to slack
+        if: failure() && github.ref == 'refs/heads/master'
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml:156]: same thing for `github.ref == 'refs/heads/master'`

File: .github/workflows/feg-workflow.yml [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+      - name: make feg precommit
+        run: |
+            go get gotest.tools/gotestsum
+            cd ${MAGMA_ROOT}/feg/gateway
+            make -C ${MAGMA_ROOT}/feg/gateway precommit
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        if: always()
+        with:
+          files: /tmp/test-results
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: Add a relevant name to the artifact like feg-precommit-test-results

File: .github/workflows/feg-workflow.yml (hunk 17 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml (hunk 18 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml:166]: fixed

File: .github/workflows/feg-workflow.yml (hunk 19 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml:156]: fixed

File: .github/workflows/feg-workflow.yml (hunk 20 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml (hunk 16 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+      - name: make feg precommit
+        run: |
+            go get gotest.tools/gotestsum
+            cd ${MAGMA_ROOT}/feg/gateway
+            make -C ${MAGMA_ROOT}/feg/gateway precommit
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        if: always()
+        with:
+          files: /tmp/feg-precommit-test-results
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: Can you add explicit name to the artifact uploaded

File: .github/workflows/feg-workflow.yml [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+      - name: run docker containers and check health
+        env:
+          DOCKER_REGISTRY: feg_
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway/docker
+            python3 build.py -e
+      - name: Tag and push to Docker Registry
+        # yamllint disable rule:line-length
+        run: |
+            ./ci-scripts/tag-push-docker.sh --images 'gateway_go|gateway_python' --tag '${GITHUB_SHA:0:8}' --tag-latest false --project feg
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: tag latest to true here too

Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: sure!

File: .github/workflows/feg-workflow.yml (hunk 14 above)
Comment (quentinDERORY) [reply .github/workflows/feg-workflow.yml:105]: Sorry to go back and forth on this but could we trigger the build part with pull request event but just not the push and notification.
TLDR: put the if statement on the tag&push and notifications steps rather than here

**Comment by quentinDERORY** [comment]
For the upload test result workflow (.github/workflows/unit-test-workflow.yml):
Add a name to the uploaded test results so we can differentiate between feg and cloud test results.
ie add the `check_name` option to the https://github.com/EnricoMi/publish-unit-test-result-action module and use the github variable  `github.event.workflow.name` .
So it looks like `feg unit tests` or 'cloud unit tests'.

File: .github/workflows/feg-workflow.yml [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+        run: |
+            go get gotest.tools/gotestsum
+            cd ${MAGMA_ROOT}/feg/gateway
+            make -C ${MAGMA_ROOT}/feg/gateway precommit
+      - name: Upload Test Results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Unit Test Results
+          path: "/tmp/feg-precommit-test-results"
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: The correct path is `/tmp/test-results`

File: .github/workflows/feg-workflow.yml [hunk 24]
```diff
@@ -0,0 +1,169 @@
...
+      - name: run docker containers and check health
+        env:
+          DOCKER_REGISTRY: feg_
+        run: |
+            cd ${MAGMA_ROOT}/feg/gateway/docker
+            python3 build.py -e
+      - name: Tag and push to Docker Registry
+        # yamllint disable rule:line-length
+        run: |
+            ./ci-scripts/tag-push-docker.sh --images 'gateway_go|gateway_python' --tag '${GITHUB_SHA:0:8}' --tag-latest true --project feg
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: use double quotes to enclose GITHUB_SHA

File: .github/workflows/feg-workflow.yml [hunk 25]
```diff
@@ -0,0 +1,169 @@
...
+      - name: Tag and push to Jfrog Registry
+        env:
+          DOCKER_REGISTRY: "${{ secrets.JFROG_DOCKER_ORC8R_REGISTRY }}"
+          DOCKER_USERNAME: "${{ secrets.JFROG_USERNAME }}"
+          DOCKER_PASSWORD: "${{ secrets.JFROG_PASSWORD }}"
+        run: |
+            ./ci-scripts/tag-push-docker.sh --images 'gateway_go|gateway_python' --tag '${GITHUB_SHA:0:8}' --tag-latest true --project feg
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: same here

File: .github/workflows/feg-workflow.yml (hunk 23 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml (hunk 24 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml (hunk 25 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml]: fixed

File: .github/workflows/feg-workflow.yml (hunk 14 above)
Comment (jwagantall) [reply .github/workflows/feg-workflow.yml:105]: I agree.. like we did in the other workflow... thank you!

Comment (quentinDERORY) [reply .github/workflows/feg-workflow.yml:105]: You can remove the if statement from there now

File: .github/workflows/feg-workflow.yml [hunk 26]
```diff
@@ -0,0 +1,170 @@
...
+          SLACK_TITLE: "Github action feg-precommit update failed"
+          SLACK_USERNAME: "Feg workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+      # Notify ci channel when push succeeds
+      - name: Notify success to slack
+        if: success() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_CHANNEL: ${SLACK_CHANNEL_ARTIFACTS}
```
Comment (quentinDERORY) [inline .github/workflows/feg-workflow.yml]: should be ${{ secrets. SLACK_CHANNEL_ARTIFACTS }} 
Can you correct it in the cloud workflow that is already merged.

File: .github/workflows/unit-test-workflow.yml [hunk 27]
```diff
@@ -29,5 +31,6 @@ jobs:
       - name: Publish Unit Test Results
         uses: EnricoMi/publish-unit-test-result-action@v1
         with:
+          check_name: github.event.workflow.name
```
Comment (quentinDERORY) [inline .github/workflows/unit-test-workflow.yml]: Can you put it between ${{ }} , just to be consistent with the line underneath

**Review by quentinDERORY — APPROVED** [review]
